src/StringParser.php

```diff
@@ -26,4 +26,17 @@ public function required()
     {
         return parent::required();
     }
+
+    /**
+     * @param string $defaultValue
+     * @return string
+     */
+    public function defaultsToIfEmpty($defaultValue)
+    {
+        if ($this->value === '') {
```

Comment:
> One question, could this be placed in AbstractValueParser?

There is no empty integer, so e.g. saying `int()->defaultsToIfEmpty(..)` makes no sense. Same applies for most of the other types.

> And perhaps empty() would be sufficient instead of === ''?

There an issue with using `empty`. It also detects e.g. `"0"` as an empty string (`empty("0") === true`). Thus we cannot use it here.